### PR TITLE
Implementation of basic leaderboard

### PR DESCRIPTION
**Pull request by martin-dj**
Implementation of leader board as outlined in issue #496.

### CODE REVIEW

**Comment by martin-dj** [comment]
Submitted as feature branch: feature/initial_leaderboard

Pull request #535

File: app/src/frontend/pages/leaderboard.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    }
+	
+	
+    renderTableData() {
+        return this.state.leaders.map((u, i) => {
+	    const username = u.username;             
+            const number_edits = u.number_edits;
+         
+	    return (
+		<tr key={i}>
```
Comment (mz8i) [inline app/src/frontend/pages/leaderboard.tsx]: the `key` prop is used by React to determine changes to the component tree. Therefore, it's best to go for something that specifically identifies the unique component - `username` here might be better than position in list.

File: app/src/frontend/pages/leaderboard.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        leaders: Leader[];
+        fetching: boolean;
+
+        //We need to track the state of the radio buttons to ensure their current state is shown correctly when the view is (re)rendered
+        number_limit: number;
+        time_limit: number;
+    }
+
+
+class LeaderBoardPage extends Component<LeaderboardProps, LeaderboardState> {
```
Comment (mz8i) [inline app/src/frontend/pages/leaderboard.tsx]: let's make the class and interface names consistent in terms of `Leaderboard`/`LeaderBoard` casing (not sure which one is more correct)

File: app/src/api/services/leaderboard.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import db from '../../db';
+
+async function getLeaders(number_limit: number, time_limit: number) {
```
Comment (mz8i) [inline app/src/api/services/leaderboard.ts]: as far as I understand, this query will be run on the logs table every time the leaderboard page is displayed. Just wondering about the impact on the server's performance? Seeing that the logs table is / will become potentially quite huge with time...

Comment (martin-dj) [reply app/src/api/services/leaderboard.ts]: I tested the implementation in my dev environment, against a database dump from the production server (dumped circa September) without issues, despite the limited resources of my virtual dev environment. 

So I would suggest merging it and seeing how it performs.

File: app/src/frontend/pages/leaderboard.tsx (hunk 1 above)
Comment (martin-dj) [reply app/src/frontend/pages/leaderboard.tsx]: Fixed

File: app/src/frontend/pages/leaderboard.tsx (hunk 2 above)
Comment (martin-dj) [reply app/src/frontend/pages/leaderboard.tsx]: Fixed

**Pull request merged**